include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -78,15 +81,68 @@ enum E_DESCRIPTOR_TYPE : uint32_t
 template<typename SamplerType>
 class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 {
+	struct SBindingRedirect
+	{
+		static constexpr inline uint32_t Invalid = ~0u;
+
+		SBindingRedirect() : bindings(nullptr), offsets(nullptr), count(0ull) {}
+
+		SBindingRedirect(const size_t _count) : count(_count)
+		{
+			bindings = std::make_unique<uint32_t[]>(count << 1);
+			offsets = bindings.get() + count;
+		}
+
+		// For a given binding number `binding` in `bindings`, return its offset in `offsets`.
+		inline uint32_t operator[](const uint32_t binding) const
+		{
+			const uint32_t index = searchForBinding(binding);
+			if (index == Invalid)
+				return Invalid;
+
+			return offsets[index];
+		}
+
+		// Returns index into the `bindings` and `offsets` array for the given binding number `binding.
+		// Assumes `bindings` is sorted.
+		inline uint32_t searchForBinding(const uint32_t binding) const
+		{
+			if (!bindings || !offsets || (count == 0ull))
+				return Invalid;
```

Comment:
I don't think this LOC will like `bindings == nullptr`
```C++
auto found = std::lower_bound(bindings.get(), bindings.get() + count, binding);
```

Reply:
ah yeah, ok `bindings` need to be checked, nut not offsets and count

Reply:
Actually, `count` should also be checked as things are now, because if the second constructor gets `_count == 0` as a param then both `bindings` and `offsets` pointers will be set to garbage i.e. both `!bindings` and `!offsets` will return true, only checking `count == 0` will save us.

Reply:
ok then check that, and put your comment in the comments

Reply:
I added an early exit in the ctor:
```C++
if (count <= 0)
    return;
```
So that we always maintain the invariant: `count==0`, `bindings==nullptr` and `offsets==nullptr`, so I can remove these checks now :)

Reply:
Resolved.